system/HTTP/ContentSecurityPolicy.php

```diff
@@ -750,6 +735,16 @@ protected function buildHeaders(ResponseInterface &$response)
 			'report-uri'      => 'reportURI',
 		];
 
+		// inject default base & default URIs if needed
+		if (empty($this->baseURI))
```

Comment:
I think your change at https://github.com/codeigniter4/CodeIgniter4/pull/1581/files#diff-340c29575fdbbbc1dd365d4e02e7ae3cR288 makes it so that `$this->baseURI` is never set, doesn't it?

Reply:
Sort of. It was originally set to 'none', and that cannot be over-ridden as that is the most restrictive that a CSP policy can be. It could start out as 'self', which can be added to (the policy allows multiple URLs as base), but none of them should start as 'none'. 
It isn't clear if the element is mandatory - doesn't appear so.
Thinking this through, and based on additional digging, I have changed the defaults for most of the *src directives.

Reply:
Well, right, but when you call `setBaseUri()` it is only adding it to the attributes array, it's not actually setting `$this->baseURI`, so this line will always result in it setting the value to self and the user's option never actually gets used, does it?

Reply:
Er, doesn't the unit test shows that if you "add" one or more baseURI origins, then it uses them, but if you don't, then it uses 'self'?

Reply:
Tests do show it working, yes, which means I think there's some dead code that's not actually being used in there regarding the $baseURI property.

Reply:
Hmmm. I don't see it :(
There are two methods not fully covered by unit testing: generateNonces() and addToHeader().
$baseURI is used in addBaseURI() and generateHeaders().
I don't see what could be "dead".

Reply:
Never mind, my bad. I wasn't following the logic all of the way through, and forgot that `addOptions` would set the baseURI property directly. Silly me. 

Then I'm out of objections. Merge away :)